File: c++/1146.cpp
//
//  main.cpp
//  test
//
//  Created by 李寻欢 on 2019/10/18.
//  Copyright © 2019 李寻欢. All rights reserved.
//
#include <stdio.h>
#include <iostream>
#include <string>
#include <map>
#include <cmath>
#include <algorithm>
#include <stack>
#include <vector>
#include <queue>
#include <unordered_map>
#include <unordered_set>
#include <string.h>
#include <set>
#include <deque>
using namespace std;
const int MAXN = 1010;
int n,m,k,num;
int indegree[MAXN]={0};
vector<int> g[MAXN];

int main(){
    scanf("%d%d",&n,&m);
    int a,b;
    for(int i=0;i<m;++i){
        scanf("%d%d",&a,&b);
        g[a].emplace_back(b);
        indegree[b]++;
    }
    scanf("%d",&k);
    vector<int> ans;
    for(int i=0;i<k;++i){
        bool flag = true;
        int temp[MAXN];
        copy(indegree, indegree+n+1, temp);
        for(int j=0;j<n;++j){
            scanf("%d",&num);
            if(temp[num] != 0) flag = false;
            for(auto &to:g[num]){
                temp[to]--;
            }
        }
        if(!flag) ans.emplace_back(i);
    }
    printf("%d",ans[0]);
    for(int i=1;i<(int)ans.size();++i) printf(" %d",ans[i]);
}

File: Python3/1146.py
n,m = map(int,input().split())
indegree = [0 for i in range(n+1)]
g = {i:[] for i in range(1,n+1)}
for _ in range(m):
    a,b = map(int,input().split())
    indegree[b] += 1
    g[a].append(b)
k=int(input())
ans = []
for i in range(k):
    topo = list(map(int,input().split()))
    temp = indegree.copy()
    flag = True
    for num in topo:
        if temp[num] != 0:
            flag = False
            ans.append(i)
            break
        for each in g[num]:
            temp[each] -= 1
for each in ans:
    print(each,end='')
    if each != ans[-1]:
        print(' ',end='')


